source/wayland/display.c

```diff
@@ -459,7 +459,7 @@ static void wayland_keyboard_modifiers(void *data, struct wl_keyboard *keyboard,
                                        uint32_t mods_locked, uint32_t group) {
   wayland_seat *self = data;
   nk_bindings_seat_update_mask(wayland->bindings_seat, NULL, mods_depressed,
-                               mods_latched, mods_locked, 0, 0, 0);
+                               mods_latched, mods_locked, group, group, group);
```

Comment:
Looking at some docs, it looks like the locked layout/group would be enough: `[...] 0, 0, group);`

https://xkbcommon.org/doc/current/group__state.html#ga04e8eac0666cc64dee6f0d5a7a773a03

and

https://wayland-book.com/seat/keyboard.html#input-events

Also tested and it works :)

Reply:
Makes sense, I guess wayland must be tracking state for the `grp:*_switch` options itself so they can be treated as just locked, and it doesn't look like latching is even possible.